main.c

```diff
@@ -104,14 +104,21 @@ static void handle_active_selection_motion(struct slurp_seat *seat, struct slurp
 
 	int32_t anchor_x = current_selection->anchor_x;
 	int32_t anchor_y = current_selection->anchor_y;
+  int32_t dist_x = current_selection->x - anchor_x;
+	int32_t dist_y = current_selection->y - anchor_y;
+  
 	current_selection->has_selection = true;
-	current_selection->selection.x = min(anchor_x, current_selection->x);
-	current_selection->selection.y = min(anchor_y, current_selection->y);
 	// selection includes the seat and anchor positions
-	current_selection->selection.width =
-		abs(current_selection->x - anchor_x) + 1;
-	current_selection->selection.height =
-		abs(current_selection->y - anchor_y) + 1;
+  int32_t width = abs(dist_x) + 1;
+	int32_t height = abs(dist_y) + 1;
+ 	if (seat->state->fixed_aspect) {
+		width = min(width, height / seat->state->aspect_ratio);
+		height = min(height, width * seat->state->aspect_ratio);
```

Comment:
I think we want `max` here, as we want the cursor to be always inside the selection, not outside.
See for example fixed ratio selection in gimp with Shift key.

![selection](https://user-images.githubusercontent.com/632767/150785906-d3692f9e-266d-4cdc-b6bd-6d0e8e2e29ad.png)

Reply:
Well, at least min does not allow to select outside screen borders. It may be a reasonable property here.